src/libguac/wol.c

```diff
@@ -85,19 +86,31 @@ static ssize_t __guac_wol_send_packet(const char* broadcast_addr,
     wol_dest.sin_family = AF_INET;
     if (inet_pton(AF_INET, broadcast_addr, &(wol_dest.sin_addr)) == 0) {
         wol_dest.sin_family = AF_INET6;
-        if (inet_pton(AF_INET6, broadcast_addr, &(wol_dest.sin_addr)) == 0)
+        if (inet_pton(AF_INET6, broadcast_addr, &(wol_dest.sin_addr)) == 0) {
+            guac_error = GUAC_STATUS_SEE_ERRNO;
+            guac_error_message = "Wake-on-LAN address does not appear to be IPv4 or IPv6";
             return 0;
+        }
```

Comment:
From the man page for `inet_pton()`:

> inet_pton() returns 1 on success (network address was successfully converted).  0 is returned if src does not contain a character string representing a valid network address in the specified address family. If af does not contain a valid address family, -1 is returned and errno is set to EAFNOSUPPORT.

[The relevant part of POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/inet_ntop.html) states similarly:

> The inet_pton() function shall return 1 if the conversion succeeds, with the address pointed to by dst in network byte order. It shall return 0 if the input is not a valid IPv4 dotted-decimal string or a valid IPv6 address string, or -1 with errno set to [EAFNOSUPPORT] if the af argument is unknown.

It doesn't look like `errno` is set for the case that `inet_pton()` returns 0. It _is_ set for -1, but that condition isn't handled here.

So, overall:

* `GUAC_STATUS_SEE_ERRNO` is probably incorrect for the case that `inet_pton()` returns 0
* We should consider handling the case that `inet_pton()` returns -1. A return value of -1 would be interpreted as success here, resulting in who-knows-what being used for the broadcast address.

Reply:
Okay, I'll refactor this a bit to handle both of those cases.

Reply:
Okay, this should be addressed, though it feels like there should be a slightly more elegant solution than this.  Played around with a `switch()` statement to see if that would work, but nothing quite as simple as `if...else`.